reorder and always show selective attribute

bigstones opened the pull request:
<!-- Raise an issue to propose your change (https://github.com/cvat-ai/cvat/issues).
It helps to avoid duplication of efforts from multiple independent contributors.
Discuss your ideas with maintainers to be sure that changes will be approved and merged.
Read the [Contribution guide](https://docs.cvat.ai/docs/contributing/). -->

<!-- Provide a general summary of your changes in the Title above -->

### Motivation and context
<!-- Why is this change required? What problem does it solve? If it fixes an open
issue, please link to the issue here. Describe your changes in detail, add
screenshots. -->

https://github.com/cvat-ai/cvat/issues/7370

and I wanted to be able to view a specific attribute on the page all at once while inspecting.

### How has this been tested?
<!-- Please describe in detail how you tested your changes.
Include details of your testing environment, and the tests you ran to
see how your change affects other areas of the code, etc. -->

#### Test reorder
setting order(text, strikethrough, underline, is_last_in_line) -> check in canvas
<img width="1762" height="1136" alt="스크린샷 2025-09-07 오후 11 44 04" src="https://github.com/user-attachments/assets/d9ac77d8-37d0-44d6-9b70-78fcfe4b6bcc" />
<img width="1762" height="1136" alt="스크린샷 2025-09-07 오후 11 44 39" src="https://github.com/user-attachments/assets/ee4904d5-cc21-41ba-8231-582118bc6004" />

setting order(text, is_last_in_line, strikethrough, underline) -> check in canvas
<img width="1762" height="1136" alt="스크린샷 2025-09-07 오후 11 44 53" src="https://github.com/user-attachments/assets/86613a69-75a1-4b33-b603-479bac60a7cc" />
<img width="1762" height="1136" alt="스크린샷 2025-09-07 오후 11 45 08" src="https://github.com/user-attachments/assets/d9c0cc64-5dca-470b-ac5b-3989db19da15" />

#### Test always show selective attribute
Non Check Enable selective object details display
<img width="1762" height="1136" alt="스크린샷 2025-09-07 오후 11 46 10" src="https://github.com/user-attachments/assets/0f2854bf-d307-4f5d-b074-ff735c4b37bb" />

Check Enable selective object details display
<img width="1762" height="1136" alt="스크린샷 2025-09-07 오후 11 46 15" src="https://github.com/user-attachments/assets/df07946a-8e0d-4611-88f6-e009298a5cb5" />

Set Attribute want to always show
<img width="1762" height="1136" alt="스크린샷 2025-09-07 오후 11 46 25" src="https://github.com/user-attachments/assets/600050a3-23a3-4ee3-ad94-d677441aefec" />


### Checklist
<!-- Go over all the following points, and put an `x` in all the boxes that apply.
If an item isn't applicable for some reason, then ~~explicitly strikethrough~~ the whole
line. If you don't do that, GitHub will show incorrect progress for the pull request.
If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [X] I submit my changes into the `develop` branch
- [x] I have created a changelog fragment <!-- see top comment in CHANGELOG.md -->
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [X] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))

### License

- [X] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

klakhov commented:
Hi @bigstones , Are you still interested in finishing this pr?
If yes, check https://github.com/cvat-ai/cvat/pull/7624 this one. We tried to implement similar feature here. I beleive we need server part here to store the attributes order. 
Also, plese clean up your PR from coverage files before review

bigstones committed:
Refactor attribute handling in label form for improved reordering

- Updated the attribute submission logic to use the current order from form values instead of fieldInstances.
- Added debug logging to track attribute order before and after reordering actions.
- Ensured re-rendering of the component after attribute reordering to reflect changes immediately.

bigstones committed:
Implement attribute reordering functionality in labels editor

- Added drag-and-drop support for reordering attributes in the label form.
- Introduced utility functions for safe array manipulation and validation.
- Enhanced the UI with visual feedback for drag handles and tooltips.
- Updated tests to cover new reordering features and ensure functionality remains intact after reordering.
- Added Cypress commands for streamlined attribute manipulation in tests.

bigstones committed:
Enhance attribute ordering in object details and canvas view

- Updated the attribute rendering logic in CanvasView to use the order defined in the label constructor, ensuring consistency with the label's attribute definitions.
- Introduced a new utility function to order attributes based on job configuration, allowing for dynamic attribute ordering in ObjectItemDetails.
- Modified the ObjectItemDetails component to utilize the new ordering function, improving the display of attributes according to the job's specifications.

bigstones committed:
Add selective display feature for object details

- Introduced selective display settings in the canvas model, allowing users to enable or disable the display of specific labels and attributes.
- Updated the canvas view to filter attributes based on the new selective display settings, ensuring only relevant details are shown.
- Added action creators and reducers to manage selective display settings in the Redux store.
- Enhanced the workspace settings modal to include options for configuring selective display preferences.
- Implemented utility functions to filter attributes and labels according to the selective display settings, improving the user experience in the annotation workflow.

bigstones committed:
Enhance selective display handling in CanvasView

- Updated the CanvasView implementation to manage selective display changes more effectively.
- Added logic to re-render text elements based on updated selective display settings, ensuring accurate representation of labels and attributes.
- Improved the condition for recreating text elements to include selective display configurations, enhancing user experience in the annotation workflow.

bigstones committed:
Refactor label form for improved attribute handling

- Removed debug logging for attribute order comparisons in the label form to streamline the submission process.
- Updated error handling comments for clarity when rendering attribute inputs.
- Enhanced comments in the attribute ordering utility to improve code readability and understanding of the mapping process.

bigstones committed:
Add selective display feature and related tests

- Introduced a selective display feature allowing users to show object details based on specific labels and attributes in workspace settings.
- Added UI controls for selecting labels and attributes, with immediate application of changes without needing to reload the workspace.
- Implemented action creators and reducers to manage selective display settings in the Redux store.
- Created comprehensive tests for actions, reducers, and utility functions related to selective display.
- Documented the feature in the changelog and user documentation for clarity on usage and functionality.

bigstones committed:
Replace arrow buttons with drag-and-drop for attribute reordering

Use react-sortable-hoc (already in project) with drag handle instead
of up/down arrow buttons for better UX.

bigstones committed:
Resolve merge conflicts with develop branch

bigstones committed:
Add display_order field for attribute ordering and clean up coverage files

- Add display_order IntegerField to AttributeSpec model with migration
- Update serializer and label form to persist attribute order
- Add Django unit tests for display_order
- Remove unused jest test files and coverage artifacts
